mypyc/ir/rtypes.py

```diff
@@ -364,6 +368,9 @@ def visit_rtuple(self, t: 'RTuple') -> str:
         parts = [elem.accept(self) for elem in t.types]
         return 'T{}{}'.format(len(parts), ''.join(parts))
 
+    def visit_rstruct(self, t: 'RStruct') -> str:
+        return "S"
```

Comment:
This is incorrect. Since all structs are mapped to a single identifier, tuples with different types will get the same name. For now it's sufficient to add `assert False` here, to not support structs in tuples.